ytelse-ung/src/main/java/no/nav/ung/sak/ytelse/ung/beregning/barnetillegg/LagBarnetilleggTidslinje.java

```diff
@@ -12,42 +12,79 @@
-    }
+    public BarnetilleggVurdering lagTidslinje(BehandlingReferanse behandlingReferanse, LocalDateTimeline<Boolean> perioder) {
+        var personopplysningGrunnlagEntitet = personopplysningRepository.hentPersonopplysninger(behandlingReferanse.getBehandlingId());
+        var barnAvSøkerAktørId = personopplysningGrunnlagEntitet.getGjeldendeVersjon().getRelasjoner()
+            .stream().filter(r -> r.getRelasjonsrolle().equals(RelasjonsRolleType.BARN))
+            .map(PersonRelasjonEntitet::getTilAktørId)
+            .toList();
 
-    static LocalDateTimeline<Barnetillegg> beregnBarnetillegg(LocalDateTimeline<Integer> antallBarnTidslinje) {
-        List<LocalDateSegment<Integer>> antallBarnMånedsvisSegmenter = antallBarnTidslinje
-            .toSegments()
-            .stream()
-            .filter(s -> s.getLocalDateInterval().contains(s.getFom().with(TemporalAdjusters.lastDayOfMonth()))) // Finner siste segment i hver måned
-            .map(s -> new LocalDateSegment<>(s.getFom().plusMonths(1).withDayOfMonth(1), TIDENES_ENDE, s.getValue())) // Mapper alle segmenter til å starte første dag i neste måned
+        var relevantPersonInfoBarn = barnAvSøkerAktørId.stream()
+            .map(this::finnRelevantPersonInfo)
             .toList();
-        var antallBarnForBarnetilleggTidslinje = new LocalDateTimeline<>(antallBarnMånedsvisSegmenter, StandardCombinators::coalesceRightHandSide);
-        return antallBarnForBarnetilleggTidslinje.mapValue(antallBarn -> new Barnetillegg(finnUtregnetBarnetillegg(antallBarn), antallBarn));
+
+        return beregnBarnetillegg(perioder, relevantPersonInfoBarn);
+
+    }
+
+    static BarnetilleggVurdering beregnBarnetillegg(LocalDateTimeline<Boolean> perioder, List<FødselOgDødInfo> relevantPersonInfoBarn) {
+        var antallBarnGrunnlagTidslinje = relevantPersonInfoBarn.stream()
+            .map(info -> new LocalDateTimeline<>(info.fødselsdato().plusMonths(1).withDayOfMonth(1), getTilDato(info), 1))
+            .reduce((t1, t2) -> t1.crossJoin(t2, StandardCombinators::sum))
+            .orElse(LocalDateTimeline.empty());
+
+        var relevantBarnetilleggTidslinje = antallBarnGrunnlagTidslinje.intersection(perioder)
+            .mapValue(antallBarn -> new Barnetillegg(finnUtregnetBarnetillegg(antallBarn), antallBarn));
+
+        return new BarnetilleggVurdering(relevantBarnetilleggTidslinje, relevantPersonInfoBarn);
     }
 
     private static int finnUtregnetBarnetillegg(Integer antallBarn) {
         return BigDecimal.valueOf(antallBarn).multiply(BARNETILLEGG_DAGSATS).intValue(); // Både sats og antall barn er heltall, så vi trenger ingen avrunding. Caster direkte til long
     }
 
+
+    private static LocalDate getTilDato(FødselOgDødInfo info) {
+        return info.dødsdato() != null ? info.dødsdato().plusMonths(1).with(TemporalAdjusters.lastDayOfMonth()) : TIDENES_ENDE;
```

Comment:
Vi bør kanskje også sjekke om barnet er under 18 år? 

Det er sikkert veldig uvanlig, men det kan jo tenkes at personen har adoptert et barn som rekker å bli 18 år før brukeren er 29

Reply:
Eg har ikkje hørt noko om at vi skal ha noko begrensning for alder på barn